plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/custom/deserializer/CacheMediatorDeserializer.java

```diff
@@ -54,22 +55,38 @@ public CacheMediator createNode(IGraphicalEditPart part, AbstractMediator object
 			setElementToEdit(mediatorModel);
 			setCommonProperties(mediator, mediatorModel);
 
-			if (StringUtils.isNotBlank(mediator.getId())) {
-				executeSetValueCommand(CACHE_MEDIATOR__CACHE_ID, mediator.getId());
-			}
-
-			if ("per-mediator".equals(mediator.getScope())) {
-				executeSetValueCommand(CACHE_MEDIATOR__CACHE_SCOPE, CacheScope.PER_MEDIATOR);
-			} else if ("per-host".equals(mediator.getScope())) {
-				executeSetValueCommand(CACHE_MEDIATOR__CACHE_SCOPE, CacheScope.PER_HOST);
-			}
 
 			if (mediator.isCollector()) {
 				executeSetValueCommand(CACHE_MEDIATOR__CACHE_ACTION, CacheAction.COLLECTOR);
 			} else {
+				String[] methods = mediator.getHTTPMethodsToCache();
+				if (methods.length != 0 && !methods[0].isEmpty()) {
```

Comment:
This is the logic we have added from the carbon-mediation side. Added the same for dev-studio as well.